engine/api/src/main/java/io/deephaven/engine/table/DataIndexTransformer.java

```diff
@@ -0,0 +1,60 @@
+package io.deephaven.engine.table;
+
+import io.deephaven.annotations.BuildableStyle;
+import io.deephaven.engine.rowset.RowSet;
+import org.immutables.value.Value;
+import org.immutables.value.Value.Immutable;
+
+import java.util.Map;
+import java.util.Optional;
+
+@Immutable
+@BuildableStyle
+public interface DataIndexTransformer {
+
+    Optional<RowSet> intersectRowSet();
+
+    Optional<RowSet> invertRowSet();
+
+    @Value.Default
+    default boolean sortByFirstRowKey() {
+        return false;
+    }
+
+    Map<ColumnSource<?>, ColumnSource<?>> keyColumnRemap();
```

Comment:
Might be worth renaming and documenting to make it very clear which side is "new" and which is "old".